react/src/Admin/components/CreateUserModal.tsx

```diff
@@ -63,40 +98,31 @@ function ErrorText(props: ErrorTextProps) {
             );
     }
 }
-// TODO: add group picker
+
 export default function CreateUserModal(props: CreateUserModalProps) {
-    const [username, setUsername] = useState("");
-    const [email, setEmail] = useState("");
-    const [isAdmin, setAdmin] = useState(false);
-    const [password, setPassword] = useState("");
-    const [confirmPassword, setConfirmPassword] = useState("");
+    const [state, dispatch] = useReducer(reducer, initState);
     const [submitting, setSubmitting] = useState(false);
     const [errorCode, setErrorCode] = useState(0);
     const [errorDetails, setErrorDetails] = useState({ error: "", message: "" });
 
+    // Reset on open/close as side effect
+    useEffect(() => {
+        dispatch({ type: "reset" });
+        setErrorCode(0);
+    }, [props.open]);
+
```

Comment:
This is slightly different than only resetting on a successful submission but it works just as well!